### DISCUSSION TITLE
How can I set gpio as persistent status?

### DISCUSSION THREAD

**Discussion by billy-tsai** [zephyrproject-rtos/zephyr#38193] [Q&A]
I want to set the GPIO to the persistent status to avoid the gpio pin status changed after the system reset. 
Does GPIO have any configuration flag that can be used for this purpose when I call gpio_pin_configure()?

**Comment by yashi** [comment]
Short answer: No.

By persistent, if you mean:
- across a reboot, no you can't.  It will be reset to the hardware reset state
- in a device driver, no I don't think we have it. We only have
  - `pin_configure()`
  - `port_get_raw()`
  - `port_set_masked_raw()`
  - `port_set_bits_raw()`
  - `port_clear_bits_raw()`
  - `port_toggle_bits()`
  - `pin_interrupt_configure()`
  - `manage_callback()`
  - `get_pending_int()`

I don't see any flag for persistence for `pin_configure()`.  But the type of `flags` is
````c
typedef uint32_t gpio_flags_t;
````
You might be able to implement your own device driver with a special flag.

ref) `zephyr/include/drivers/gpio.h`
